python/oneseismic/scan/scanners.py

```diff
@@ -206,3 +206,133 @@ def report(self):
+    crossline direction.
+
+    Note that this assumes uniform inline and crossline spacing.
+
+    """
+    def __init__(self, il_word, xl_word, endian):
+        super().__init__(endian)
+        self.xl_word = xl_word
+        self.il_word = il_word
+
+        self.iline_min = None
+        self.iline_max = None
+        self.xline_min = None
+        self.xline_max = None
+
+        self.p0_linenos = None
+        self.p1_linenos = None
+        self.p2_linenos = None
+
+        self.p0 = None
+        self.p1 = None
+        self.p2 = None
+
+    def get_cdp(self, header, word):
+        scale = header[segyio.su.scalco]
+        # SEG-Y specifies that a scaling of 0 should be interpreted as identity
+        if scale == 0:
+            scale = 1
+
+        cdp = header[word]
+        if scale > 0:
+            return cdp * scale
+        else:
+            return cdp / -scale
+
+    def scan_trace_header(self, header):
+        iline = self.intp.parse(header[self.il_word])
+        xline = self.intp.parse(header[self.xl_word])
+        cdpx = self.get_cdp(header, segyio.su.cdpx)
+        cdpy = self.get_cdp(header, segyio.su.cdpy)
+
+        if self.iline_min is None or self.iline_min > iline:
+            self.iline_min = iline
+
+        if self.iline_max is None or self.iline_max < iline:
+            self.iline_max = iline
+
+        if self.xline_min is None or self.xline_min > xline:
+            self.xline_min = xline
+
+        if self.xline_max is None or self.xline_max < xline:
+            self.xline_max = xline
+
+        closer_to_origo = \
+            self.p0_linenos is not None and \
+            is_closer(
+                (self.iline_min, self.xline_min),
+                (iline, xline),
+                self.p0_linenos
+            )
+
+        if self.p0_linenos is None or closer_to_origo:
+            self.p0_linenos = (iline, xline)
+            self.p0 = (cdpx, cdpy)
+
+        closer_to_il_corner = \
+            self.p1_linenos is not None and \
+            is_closer(
+                (self.iline_max, self.xline_min),
+                (iline, xline),
+                self.p1_linenos
+            )
+
+        if self.p1_linenos is None or closer_to_il_corner:
+            self.p1_linenos = (iline, xline)
+            self.p1 = (cdpx, cdpy)
+
+        closer_to_xl_corner = \
+            self.p2_linenos is not None and \
+            is_closer(
+                (self.iline_min, self.xline_max),
+                (iline, xline),
+                self.p2_linenos
+            )
+
+        if self.p2_linenos is None or closer_to_xl_corner:
+            self.p2_linenos = (iline, xline)
+            self.p2 = (cdpx, cdpy)
+
+
+    def report(self):
+        M = np.array([
+            [self.p0_linenos[0], self.p0_linenos[1], 1],
+            [self.p1_linenos[0], self.p1_linenos[1], 1],
+            [self.p2_linenos[0], self.p2_linenos[1], 1]
+        ])
+
+        linos_to_cdpx, linos_to_cdpy = np.linalg.lstsq(
```

Comment:
I don't think that is possible since our system is not over determined.

Reply:
That system actually has infinitely many solutions :D Numpy will select one based on some criteria (smallest 2-norm |x| according to the docs). It will only throw if it does not converge. Should not happen for over determined systems either, so not sure what could potentially cause that.

Reply:
![image](https://user-images.githubusercontent.com/44965282/152532154-30307a66-123c-4035-9cdf-bb45d096c2fa.png)
Everything can cause that, in theory. You don't know if system of linear equations is solvable before you do something with it.

But good to know that numpy will choose one for the case when many solutions are there.

Reply:
I will use a different approach that throws for inconsistent systems and linearly dependent points. Additionally I will add some sanity checks